alicloud/resource_alicloud_instance.go

```diff
@@ -1231,6 +1245,26 @@ func modifyInstanceType(d *schema.ResourceData, meta interface{}, run bool) (boo
 			return update, err
 		}
 		d.SetPartial("instance_type")
+		// Ensure instance's type has been replaced successfully.
+		timeout := DefaultTimeoutMedium
+		for {
+			instance, err := ecsService.DescribeInstance(d.Id())
+
+			if err != nil {
+				return update, WrapError(err)
+			}
+
+			if instance.InstanceType == d.Get("instance_type") {
```

Comment:
I think the line 1257 should be `if instance.InstanceType == d.Get("instance_type").(string)`